### PR TITLE
feat(platform): added id and domain to projects

### PR DESCRIPTION
**Pull request by thsig**
**What this PR does / why we need it**:

This replaces the temporary setup we had based on project names (using ids instead) and the `GARDEN_CLOUD` env variable.

### CODE REVIEW

File: garden-service/src/config/project.ts [hunk 1]
```diff
@@ -277,6 +278,12 @@ export const projectDocsSchema = () =>
         .meta({ internal: true })
         .description("The path to the project config file."),
       name: projectNameSchema(),
+      // TODO: Refer to platform documentation for more details.
+      id: joi.string().description("The project's ID in Garden Cloud."),
```
Comment (edvald) [inline garden-service/src/config/project.ts:282]: I suggest marking these with `.meta({ internal: true })` for the time being.

File: garden-service/src/config/project.ts [hunk 2]
```diff
@@ -277,6 +278,12 @@ export const projectDocsSchema = () =>
...
+      // TODO: Refer to platform documentation for more details.
+      domain: joi
+        .string()
+        .description("The domain to use for cloud features. Should point to the API/backend endpoint."),
```
Comment (edvald) [inline garden-service/src/config/project.ts:286]: endpoint -> hostname?

File: garden-service/src/garden.ts [hunk 3]
```diff
@@ -287,20 +289,38 @@ export class Garden {
...
+          errorMessages.push(deline`
+            ${chalk.bold("project.domain")} is not set in your project-level ${chalk.bold("garden.yml")}. Make sure it
+            is set to the appropriate API backend endpoint (e.g. myusername-cloud-api.cloud.dev.garden.io, without an
+            http/https prefix).
+          `)
+        }
+        if (!projectId) {
+          errorMessages.push(deline`
+            ${chalk.bold("project.id")} is not set in your project-level ${chalk.bold("garden.yml")}. Please use the
+            ${chalk.bold("garden set project")} command to set this value, or paste it from the platform's web UI
```
Comment (eysi09) [inline garden-service/src/garden.ts:309]: Is it weird to suggest the command before adding it? But I suppose you can say that about the other stuff as well.

Comment (thsig) [reply garden-service/src/garden.ts:309]: Yeah, agreed. I'll remove the reference to the command in here (was planning to implement the command at the time I wrote that description).

**Pull request merged**